src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracks.java

```diff
@@ -39,60 +47,52 @@
     private static final int TOWER_OFFSET = 8;
 
     /**
-     * Constructor for the Barracks building.
+     * Constructor for a AbstractBuilding.
      *
-     * @param c Colony the building is in.
-     * @param l Location of the building.
+     * @param colony Colony the building belongs to.
+     * @param pos    Location of the building (it's Hut Block).
      */
-    public BuildingBarracks(final Colony c, final BlockPos l)
+    protected BuildingBarracks(@NotNull final Colony colony, final BlockPos pos)
     {
-        super(c, l);
+        super(colony, pos);
     }
 
-    /**
-     * Gets the name of the schematic.
-     *
-     * @return Barracks schematic name.
-     */
-    @NotNull
     @Override
     public String getSchematicName()
     {
-        return BARRACKS;
+        return SCHEMATIC_NAME;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)